packages/central-server/src/apiV2/surveys/assertDataElementMustMatchDataGroupServiceType.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+
+/**
+ *
+ * @param {string} dataElementServiceType
+ * @returns {boolean}
+ *
+ * @description This is a utility function to check whether the data element can have a different service type to the data group it is in. Tupaia data elements can have dhis or Tupaia data groups. While the logic here is simple, keeping it in one place will reduce the likelihood of bugs.
+ */
+export const assertDataElementMustMatchDataGroupServiceType = dataElementServiceType => {
```

Comment:
Good call to consolidate this piece of logic into its own function. Whenever there's a business rule like this its always nice if it's just defined in one place.

```suggestion
export const dataElementMustMatchDataGroupServiceType = dataElementServiceType => {
```
Just a little rename suggestion, typically `assert` functions will throw an error if the assertion fails. This is more of a check